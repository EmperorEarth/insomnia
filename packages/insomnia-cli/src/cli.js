// @flow

import { program } from 'commander';
import { makeGenerateCommand } from './commands/generate';
import * as packageJson from '../package.json';

export async function go() {
  await program
    .storeOptionsAsProperties(false)
    .passCommandToAction(false)
    .version(packageJson.version, '-v, --version')
    .addCommand(makeGenerateCommand())
    .parseAsync(process.argv);
}
