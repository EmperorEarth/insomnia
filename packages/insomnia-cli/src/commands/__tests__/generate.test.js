import * as cli from '../../cli';
import execa from 'execa-wrap';

const initInso = () => {
  return args => {
    const cliArgs = `node test ${args}`.split(' ');
    // console.log('calling cli.go with: %o', cliArgs);
    return cli.go(cliArgs);
  };
};

describe('handleGenerateConfig', () => {
  let inso = initInso();
  beforeEach(() => {
    inso = initInso();
  });

  it('should warn when --type option is not recognized', async () => {
    const result = await execa('./bin/inso');

    expect(result).toMatchSnapshot();
  });

  xit('should warn when --type option is not recognized', () => {
    inso('generate config -t test file.yaml');
  });
});
