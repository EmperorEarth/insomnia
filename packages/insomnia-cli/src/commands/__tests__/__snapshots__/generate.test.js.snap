// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`handleGenerateConfig should warn when --type option is not recognized 1`] = `
"
  command: ./bin/inso
  code: 1
  failed: true
  killed: false
  signal: null
  timedOut: false

  stdout:
  -------
  Usage: inso [options] [command]

  A CLI for Insomnia!

  Options:
    -v, --version   output the version number
    -h, --help      display help for command

  Commands:
    generate        Code generation utilities
    help [command]  display help for command
  -------
  stderr:
  -------
  
  -------
  "
`;
